### ISSUE TITLE
Fetch abortation for file urls does not work

### ISSUE THREAD

**Issue by ferjm** [servo/servo#22510]
I still see `process_response_chunk` calls coming after cancelling a fetch request to a file:// url.

**Comment by jdm** [comment]
It might be true that we don't yet support cancellation, but since we're using asynchronous communication we also cannot guarantee that no more response chunks will be received after cancelling a fetch. Consumers need to store some kind of state that allows them to ignore unwanted data.

**Comment by ferjm** [comment]
Yeah, I don't expect to receive zero response chunks after cancellation, but to stop receiving them at some point before EOF :). That's what happens with fetch cancellation for http urls.

**Comment by jdm** [comment]
That still might not be a reasonable expectation, since the file reading will be streaming chunks across the channel as fast as it can read them.
